Add status code to rest.suppressed log output

cbuescher opened the pull request:
Currently the rest.suppressed logger in RestResponse logs the request path and the request parameters for 500 errors (warn level) and 400s (debug). In order to be able to filter on those status codes more efficiently we should add them to the log message.
This change also adds testing around the logic deciding when we log with respect to the request parameters.

cbuescher committed:
Add status code to rest.suppressed log output

Currently the rest.suppressed logger in RestResponse logs the request path and
the request parameters for 500 errors (warn level) and 400s (debug). In order to
be able to filter on those status codes more efficiently we should add them to
the log message.

cbuescher committed:
Update docs/changelog/100990.yaml

elasticsearchmachine commented:
Hi @cbuescher, I've created a changelog YAML for you.

elasticsearchmachine commented:
Pinging @elastic/es-search (Team:Search)

elasticsearchmachine commented:
Pinging @elastic/es-core-infra (Team:Core/Infra)

cbuescher commented:
Just as a note, as a follow up we'd like to remove the condition to not log when the "error_trace" parameter is set by the client, in order to have a full record of errors in the logs for monitoring.

cbuescher committed:
Update 100990.yaml

quux00 commented on server/src/main/java/org/elasticsearch/rest/RestResponse.java:
```diff
@@ -120,9 +120,10 @@ public RestResponse(RestChannel channel, RestStatus status, Exception e) throws
             Supplier<?> messageSupplier = () -> String.format(
                 Locale.ROOT,
-                "path: %s, params: %s",
+                "path: %s, params: %s, status: %s",
```
Minor nit: should we use `%d` formatter instead of `%s` since status is an int?

cbuescher committed:
Update RestResponse.java

cbuescher commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'main' into add-status-to-rest-suppressed

pgomulka reviewed (approved):
LGTM

cbuescher committed:
Log errors in RestResponse regardless of error_trace parameter

Currently we only log exceptions in RestResponse when the request parameter
"error_trace" isn't set or set to "false". If the client requests traces in the
rest response, we skip the server side logging. In order to improve visibility
of those errors in our logs this changes the logging behaviour to only depend on
the "rest.exception.stacktrace.skip" which is only used internally.

Closes #100884

cbuescher committed:
Revert "Log errors in RestResponse regardless of error_trace parameter"

This reverts commit 94e6c6864bf21c46b8077055ead3283226d4475f.

quux00 reviewed (approved):
LGTM
